### PR TITLE
Decaffeinate!

### PR DESCRIPTION
**Pull request by zeke**
This PR migrates the codebase from CoffeeScript to JavaScript.

Here's what I did:

1. Changed `name` in package.json from `liquid-node` to `liquid`
1. I ran `npx decaffeinate test` to convert all the test files from `.coffee` to `.js`
1. I tried to use [`decaffeinate`](https://ghub.io/decaffeinate) to also convert the source code in `src` too, but it didn't really work out. The files converted without issue, but the changes caused over 100 tests to fail. So instead I backed out that change and ran `npm run compile` to compile the CoffeeScript to JavaScript.
1. I removed the `src` directory and all the coffescript
1. I took the `lib` directory out of gitignore so it's part of the repo
1. I added [`standard`](https://ghub.io/standard) as a dev dependency and used `standard --fix` to clean up lots of issues in the compiled `lib/**/*.js` files.
1. I manually cleaned up oddities from the coffee-generated code, one at a time, making sure to keep the test suite green.

There's more to be done before the code in the `lib` directory is super-nice and free of all coffee remnants, but it works and the existing test suite is still passing. 💅 

I'd be in favor of doing the following:

- [ ] land this PR
- [ ] publish it to GitHub and npm as a new major version `liquid@2` (Edit: Maybe `liquid@4`?)
- [ ] run `npm deprecate liquid-node@"*" "'liquid-node' has been renamed to 'liquid'"
- [ ] continue to make improvements to the source code and documentation in followup PRs.

@sirlantis @sarahs how does that sound?

Resolves https://github.com/sirlantis/liquid-node/issues/70

### CODE REVIEW

**Comment by zeke** [comment]
Forgot to mention: I also updated the Travis config to run on Node 8 and 10

**Comment by zeke** [comment]
And updated `engines.node` to `>=8`

**Comment by zeke** [comment]
🤔 Maybe we should actually publish `liquid@4`, which would make more sense as a followup to `liquid-node@3`

```
$ npm show liquid-node dist-tags
{ latest: '3.0.0' }
```

**Comment by zeke** [comment]
We should also probably publish a final version of `liquid-node` as `3.0.1` with a README update mentioning the rename. Otherwise anyone who visits the old package on the npm website won't easily discover that the package has been renamed.

File: lib/liquid/drop.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+module.exports = (function () {
+  function Drop () {}
```
Comment (sirlantis) [inline lib/liquid/drop.js]: I guess we should convert some of these into proper ES classes at a future point in time.

**Comment by zeke** [comment]
@sarahs to be sure these changes work on our internal project, I ran `npm uninstall liquid-node && npm install zeke/liquid-node#decaffeinate && npm test`, changed the `require('liquid-node')` to `require('liquid')`, and all tests are still passing!

File: package.json [hunk 2]
```diff
@@ -18,45 +18,45 @@
...
-    "coffee-script": "~1.10.0",
-    "coffeelint": "^1.11.1",
     "coveralls": "^2.11.4",
     "jscoverage": "^0.6.0",
     "mocha": "~2.5.3",
     "mocha-lcov-reporter": "1.2.0",
     "sinon": "^1.16.1",
-    "sinon-chai": "^2.8.0"
+    "sinon-chai": "^2.8.0",
+    "standard": "^12.0.1",
```
[the remark this answers is not in the document]
Comment (zeke) [reply package.json]: Maybe we could do https://github.com/sheerun/prettier-standard -- best of both worlds?

**Pull request merged**